checkups/kubevirt-vm-latency/vmlatency/internal/checkup/checkup.go

```
@@ -107,6 +119,23 @@ func (c *checkup) Run() error {
 }
 
 func (c *checkup) Teardown() error {
+	const errMessagePrefix = "teardown"
+
+	ctx, cancel := context.WithTimeout(context.Background(), c.teardownTimeout)
+	defer cancel()
+
+	var errs []string
+	if err := c.client.DeleteVmis(ctx, c.sourceVM, c.targetVM); err != nil {
+		errs = append(errs, err.Error())
+	}
+	if err := c.client.WaitVmisDispose(ctx, c.sourceVM, c.targetVM); err != nil {
+		errs = append(errs, err.Error())
+	}
+
+	if len(errs) > 0 {
+		return fmt.Errorf("%s: %s", errMessagePrefix, strings.Join(errs, ", "))
+	}
```

Please see PR #52, it has something similar.

---

Note that in #52 at the end `[]string` was expected, here we need to return an error.